inc/lift/RequestHandle.h

```diff
@@ -38,13 +39,15 @@ class RequestHandle {
 
 private:
     RequestHandle(
-        RequestPool* request_pool,
+        RequestPool& request_pool,
         std::unique_ptr<Request> request_handle);
-
-    /// The request pool that owns this request.
-    RequestPool* m_request_pool;
-    /// The actual underlying request object.
-    std::unique_ptr<Request> m_request_handle;
+    
+    explicit RequestHandle(std::shared_ptr<SharedRequest> shared_request);
+    
+    [[nodiscard]]
+    auto createSharedRequestOnHeap() const -> std::unique_ptr<std::shared_ptr<SharedRequest>>;
+    
+    std::shared_ptr<SharedRequest> m_shared_request;
```

Comment:
Does this need to be thread safe?

Reply:
Kind of. What is the issue you're concerned about?

Reply:
Multiple threads accessing the same shared_ptr object is _not_ thread safe

Reply:
Ah, okay, I understand what you meant now.
This does not need to be thread safe, because only one thread at a time will access this RequestHandle/shared pointer object.